### PR TITLE
Add explicit envvar to control if we mask /proc/driver/nvidia/params

### PR DESCRIPTION
**Pull request by klueska**

### CODE REVIEW

File: deployments/helm/k8s-dra-driver/values.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ selectorLabelsOverride: {}
 allowDefaultNamespace: false
 
 deviceClasses: ["gpu", "mig", "imex"]
+maskNvidiaDriverParams: false
```
Comment (elezar) [inline deployments/helm/k8s-dra-driver/values.yaml:38]: do we want to add a description to the value to call out that this is for testing under kind with multiple nodes?

Comment (klueska) [reply deployments/helm/k8s-dra-driver/values.yaml:48]: updated

Comment (elezar) [reply deployments/helm/k8s-dra-driver/values.yaml:48]: I would assume that a user would first inspect the values file to determine what can be set and not necessarily go as deep as the template file.

I would summarise it as follows: the comment in the template file is for us, and the comment in the values file is for users.

> What is your preference?

I prefer grouping options for readability, but I can also see that doing this early is not ideal -- especially if we don't have something that makes sense as a logical unit. Let's leave it top-level for now.

**Review by elezar — APPROVED** [review]
I think my remaining comments are minor.

Thanks @klueska.

File: deployments/helm/k8s-dra-driver/templates/kubeletplugin.yaml [hunk 2]
```diff
@@ -74,6 +76,8 @@ spec:
         resources:
           {{- toYaml .Values.kubeletPlugin.containers.plugin.resources | nindent 10 }}
         env:
+        - name: MASK_NVIDIA_DRIVER_PARAMS
+          value: "{{ .Values.maskNvidiaDriverParams }}"
```
[the remark this answers is not in the document]
Comment (klueska) [reply deployments/helm/k8s-dra-driver/templates/kubeletplugin.yaml:75]: I don't think its necessary. If / when we do that we should do it for all options to be symmetrical.

File: deployments/helm/k8s-dra-driver/values.yaml (hunk 1 above)
Comment (klueska) [reply deployments/helm/k8s-dra-driver/values.yaml:48]: Updated the comment in the template to be more geared towards us rather than the user.

**Pull request merged**